test/close.test.js

```diff
@@ -233,3 +233,37 @@ t.test('Current opened connection should not accept new incoming connections', t
     })
   })
 })
+
+test('Cannot be reopened the closed server without listen callback', t => {
+  t.plan(2)
+  const fastify = Fastify()
+
+  fastify.listen(0)
+    .then(() => {
+      fastify.close()
+        .then(() => {
+          fastify.listen(0).catch(err => {
+            t.ok(err)
+            t.is(err.code, 'FST_ERR_REOPENED_CLOSE_SERVER')
+          })
+        })
+    })
+})
```

Comment:
would you mind to use async/await in the test?

Reply:
I got it